### PR TITLE
Add optional support for uploading Python wheels.

### PR DESCRIPTION
**Pull request by mauritsvanrees**
Use the new zest.releaser[recommended] extra, or pip install
wheel yourself next to zest.releaser.

Issue #55

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
Note that I rebased this branch on master, to get my recent test fix in.

I did a rebase and a forced push, as advocated in https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request which showed up in a Google search.
I was skeptical of the forced push, but thought I would try it just this once. Indeed it gives problems when you have already checked out the branch somewhere else and you do a pull, giving you conflicts. So a merge would have been better.
If you have already checked out this branch locally, you may want to remove it before pulling.

For clarity: a merge of this pull request to master would be safe.

**Comment by reinout** [comment]
Yeah, I always merge. Rebasing sounds fun, but there shouldn't be a copy of the branch anywhere else. Otherwise you're asking for a medieval treatment :-)

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,15 @@ Changelog for zest.releaser
 3.57 (unreleased)
 -----------------
 
+- Add optional support for uploading Python wheels.  Use the new
+  ``zest.releaser[recommended]`` extra, or pip install ``wheel``
+  yourself next to ``zest.releaser``.  Create or edit ``setup.cfg`` in
```
Comment (reinout) [inline CHANGES.rst]: Needs a verb somewhere. Now I apparently should "pip install" myself :-)

**Comment by reinout** [comment]
Change looks good, apart from one thing: shouldn't wheels be on by default (provided that the wheel package is available)? At least, I can imagine it being the default in 2016. Won't changing it then be hard?

I haven't used wheels much, so I don't know much about the drawbacks.

**Comment by mauritsvanrees** [comment]
I have not used wheels at all. (And zc.buildout cannot handle them yet, so I won't be using them much in the near future.) I have my info from http://pythonwheels.com. From what I read there, I distill this for deciding whether wheels are good for a package:

- pure python 2 or python 3 package: no problem

- using 2to3 for Python 3 support: a problem, though I do not see there how you should then handle this, if you should maybe forget about wheels for now.

- Using other ways to support both Python 2 and 3: no problem, you should upload a *universal* wheel.

- Package with optional C extensions: do NOT upload a univeral wheel. non-universal is presumably fine.

- Package with C extensions in general (optional or default): upload wheel only for Mac or Windows.

That sounded a bit tricky for zest.releaser to decide on its own, so I gave the user an explicit way to indicate this for a package.

Maybe we can come up with something better.

File: doc/source/uploading.rst [hunk 2]
```diff
@@ -89,6 +89,19 @@ You can use no/false/off/0 or yes/true/on/1 as answers; upper, lower or mixed
 case are all fine.
 
 
+Uploading wheels
+----------------
+
+First, you should install the ``zest.releaser[recommended]`` extra, or
+``pip install wheel`` yourself next to ``zest.releaser``.  Then create
```
Comment (reinout) [inline doc/source/uploading.rst]: Same textual change needed.

**Comment by reinout** [comment]
Ok, clear explanation. Sounds good to leave it optional for now

**Comment by reinout** [comment]
Merge at will.

Can you take a quick peek at #82 and perhaps merge that one as well (in case you know a bit more than me about tox.ini).

And afterwards do a release?

**Pull request merged**